### ISSUE TITLE
Rule `no-unsafe-spy` is failed when `jasmine.createSpy` is used in function which is defined outside but called inside before/after/it

### ISSUE THREAD

**Issue by Valdermeyder** [tlvince/eslint-plugin-jasmine#111]
**Do you want to request a *feature* or report a *bug*?**
I suppose it is rather new *feature* than *bug*

**What is the current behavior?**
```javascript
describe('suite', function () {
	beforeEach(function () {
		initSpies();
	});
});

function initSpies() {
	return {
		params: {},
		getData: jasmine.createSpy('getData'),
		getValidations: jasmine.createSpy('getValidations')
	};
}
```
For code above next warning will be shown `warning  Spy declared outside of before/after/it block  jasmine/no-unsafe-spy`

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
Warning should not be shown if function `initSpies` is called only from before/after/it block

**Please mention your node.js, eslint-plugin-jasmine and operating system version.**
node.js v7.2.1
eslint-plugin-jasmine v2.2.0

**Comment by sebastianzillessen** [comment]
Hello! I have the same issue:

    describe("someTest", ()=>{
        it("test1", ()=>{
            createSpy();
        })
        it("test2", ()=>{
            createSpy();
        })
    })
    function createSpy(){
        return jasmine.createSpy('validate() spy')
    }

Causes a 

    warning  Spy declared outside of before/after/it block  jasmine/no-unsafe-spy

Warning.

**Comment by kazooiebombchu** [comment]
I'm also faced the same problem when I tried to declare helper function which will return a spy. I'm ended up disabling this rule for now.

**Comment by GreenGremlin** [comment]
Same here. Any ideas on a possible fix for this rule?

**Comment by Kazhuu** [comment]
I just disabled because couldn't figure out any other way to fix this issue at the moment.

**Comment by guilhermetod** [comment]
I don't, think there's much this plugin can do, as it can't preview how this function is going to be used, unless it keeps a track of everytime the function is called.

The alternatives are 1) obviously, disable the rule for the function block and 2) the following:

```js
describe('suite', () => {
  let initSpies;
  let params;
  let getData;
  let getValidations;

  beforeAll(() => {
    initSpies = () => ({
      params: {},
      getData: jasmine.createSpy('getData'),
      getValidations: jasmine.createSpy('getValidations'),
    });
  });

  beforeEach(() => {
    initSpies();
  });
});
```

**Comment by guidoderooij** [comment]
@tlvince I face the same issue. If you want I can contribute to this, I already have working code including tests.

My fix works as follows: Defining a spy inside a function is allowed as long as that function is called from inside a before/after/it block. If the function is called outside an error is reported: 'Function containing a spy is called outside of before/after/it block'

This would also fix issue #344
